src/tm-threads.c

```diff
@@ -1842,6 +1843,64 @@ void TmThreadContinueThreads()
     return;
 }
 
+void TmThreadEnsureUnpaused()
+{
+    ThreadVars *tv = NULL;
+    int i = 0;
+    int paused_threads = 0;
+
+    SCMutexLock(&tv_root_lock);
+
+    do {
+        paused_threads = 0;
+        for (i = 0; i < TVT_MAX; i++) {
+            tv = tv_root[i];
+            while (tv != NULL) {
+                if (TmThreadsCheckFlag(tv, THV_PAUSED)) {
+                    paused_threads++;
+                }
+                tv = tv->next;
+            }
+        }
+        if (paused_threads) {
+            SCLogDebug("%u threads paused; waiting 100us for them to unpause", paused_threads);
+            SleepUsec(100);
```

Comment:
Sleeping -- even for a short time -- with the mutex held should be avoided.

Suggest releasing the lock, sleeping, reacquire the lock, and then rescan here and in `EnsureRunning`